momepy/diversity.py

```diff
@@ -899,19 +899,21 @@ def __init__(
         elif weighted is None:
             data = data.set_index(unique_id)[values]
 
+            if NumpyVersion(np.__version__) >= "1.22.0":
+                method = dict(method=interpolation)
+            else:
+                method = dict(interpolation=interpolation)
             for index in tqdm(data.index, total=data.shape[0], disable=not verbose):
                 if index in spatial_weights.neighbors.keys():
                     neighbours = [index]
                     neighbours += spatial_weights.neighbors[index]
-
                     values_list = data.loc[neighbours]
-                    if NumpyVersion(np.__version__) >= "1.22.0":
-                        method = dict(method=interpolation)
-                    else:
-                        method = dict(interpolation=interpolation)
-                    results_list.append(
-                        np.nanpercentile(values_list, percentiles, **method)
-                    )
+                    all_nan = np.isnan(values_list).all()
+                    if not all_nan:
+                        values_list = np.nanpercentile(
+                            values_list, percentiles, **method
+                        )
+                    results_list.append(values_list)
```

Comment:
What happens when `all_nan == True`? You append an array of an arbitrary length full of NaNs to `results_list`, instead of an array of length == `len(percentiles)` as `nanpercentiles` itself returns (I assume). I don't think that is the expected behaviour.

Reply:
Ah! Good catch! How about this:

```python
import numpy
percentiles = [25, 75]
value_arrays = [
    numpy.array([[10, 5, 0]]),                        # all numeric
    numpy.array([[10, numpy.nan, 0]]),                # not all NaNs
    numpy.array([[numpy.nan, numpy.nan, numpy.nan]])  # all NaNs
]
nan_filler = numpy.empty(len(percentiles))
nan_filler[:] = numpy.nan
for a in value_arrays:
    a_all_nan = numpy.isnan(a).all()
    if a_all_nan:
        values_list = nan_filler
    else:
        values_list = numpy.nanpercentile(a, percentiles, method="midpoint")
    print(values_list)
# [2.5 7.5]       # return percentiles of `a` object
# [5. 5.]         # return original `a` object
# [nan nan]       # return array length len(percentiles) filled with NaNs
```